server/command.go

```diff
@@ -596,6 +605,53 @@ func (p *Plugin) subscribeCommand(ctx context.Context, parameters []string, chan
 		return invalidSubscribeSubCommand
 	}
 }
+func (p *Plugin) pipelinesCommand(ctx context.Context, parameters []string, channelID string, info *gitlab.UserInfo) string {
+	if len(parameters) == 0 {
+		return invalidPipelinesSubCommand
+	}
+	subcommand := parameters[0]
+	switch subcommand {
+	case commandRun:
+		if len(parameters) < 2 {
+			return specifyRepositoryMessage
+		}
+		namespace := parameters[1]
+		ref := parameters[2]
```

Comment:
We're expecting 3 params here so we need to assert that there are 3

```suggestion
		if len(parameters) < 3 {
			return specifyRepositoryMessage
		}
		namespace := parameters[1]
		ref := parameters[2]
```